### PR TITLE
Introduces SignedBlock for the mockchain.

### PR DESCRIPTION
**Pull request by qnikst**

### CODE REVIEW

File: chain-impl-mockchain/src/block.rs [hunk 1]
```diff
@@ -62,6 +64,98 @@ impl property::HasTransaction<SignedTransaction> for Block {
...
+
+/// `Block` is an element of the blockchain it contains multiple
+/// transaction and a reference to the parent block. Alongside
+/// with the position of that block in the chain.
+#[derive(Debug, Clone, PartialEq, Eq, Deserialize, Serialize)]
+pub struct SignedBlock {
+    /// Internal block.
+    block: Block,
+    /// List of cryptographic signatures that verifies the block.
+    signatures: Vec<(Address, Signature)>,
```
Comment (NicolasDP) [inline chain-impl-mockchain/src/block.rs]: there is only one signature in our case, no need for address.

we just need:

```rust
struct SignedBlock {
    block: Block,
    signatures: Signature,
    public_key: PublicKey,
}
```

File: chain-impl-mockchain/src/block.rs [hunk 2]
```diff
@@ -62,6 +64,98 @@ impl property::HasTransaction<SignedTransaction> for Block {
...
+}
+
+impl SignedBlock {
+    /// Create a new signed block.
+    pub fn new(block: Block) -> Self {
+        SignedBlock {
+            block: block,
+            signatures: vec![],
+        }
```
Comment (NicolasDP) [inline chain-impl-mockchain/src/block.rs]: no, so what you need to do here (once the type is right) is:

```rust
impl SignedBlock {
    pub fn new(key: &PrivateKey, block: Block) -> Self {
        let id = block.id();
        let signature = key.sign(id.as_ref());
        Block { block, signature, public_key: key.public() }
    }
```

**Comment by qnikst** [comment]
@NicolasDP please take another look, I've addressed your comments.

File: chain-impl-mockchain/src/block.rs [hunk 3]
```diff
@@ -69,6 +70,80 @@ impl property::HasTransaction<SignedTransaction> for Block {
...
+    type Date = <Block as property::Block>::Date;
+    type Header = <Block as property::Block>::Header;
+
+    /// Identifier of the block, currently the hash of the
+    /// serialized transaction.
+    fn id(&self) -> Self::Id {
+        self.block.id()
+    }
+
+
```
Comment (NicolasDP) [inline chain-impl-mockchain/src/block.rs]: rustfmt is going to fail on this line. please remove.

Comment (qnikst) [reply chain-impl-mockchain/src/block.rs]: Fixed

**Pull request merged**